nextcord/http.py

```diff
@@ -237,28 +616,44 @@ async def request(
         *,
         files: Optional[Sequence[File]] = None,
         form: Optional[Iterable[Dict[str, Any]]] = None,
+        auth: Optional[str] = MISSING,
+        retry_request: bool = True,
         **kwargs: Any,
     ) -> Any:
-        bucket = route.bucket
-        method = route.method
-        url = route.url
-
-        lock = self._locks.get(bucket)
-        if lock is None:
-            lock = asyncio.Lock()
-            self._locks[bucket] = lock
-
-        # header creation
-        headers: Dict[str, str] = {
-            "User-Agent": self.user_agent,
-        }
+        """|coro|
+
+        Makes an API request to Discord, handling authorization (if needed), rate limits, and limited error handling.
+
+        Parameters
+        ----------
+        route: :class:`Route`
+            The Discord Route to make the API request for.
+        files: Optional[Sequence[:class:`File`]]
+            pass
+        form: Optional[Iterable[:class:`dict`[:class:`str`, `Any`]]]
+            pass
+        auth: :class:`str` | `None` | `MISSING`
+            Authorization string to use. Will not auto-format given tokens. For example, a bot token must be provided
+            as "Bot <bot token>". If set to `None`, no authorization will be used. If left unset, the default
+            auth will be used. (if the default is not set, no auth will be used.)
+        retry_request: :class:`bool`
+            If the request should be retried in specific cases. This mainly concerns 500 errors (Discord server issues)
+            or 429s. (ratelimit issues)
+        kwargs
+            This is purposefully undocumented. Behavior of extra kwargs may change in a breaking way at any point, and
```

Comment:
Currently yes, but I don't want behavior of additional kwargs to be defined or relied on in any way. Mentioning that additional kwargs are simply routed to `ClientSession.request` implies that it may stay that way.

Reply:
Fair point. In that case, I think some of the important kwargs that should (hopefully) stay unchanged like `json` and `query` should be documented.

Reply:
When webhooks get moved over to using HTTPClient, that would probably be a good time to expand on that.

Reply:
This needs to be somewhat documented (for example the json param) at least.